actpool/queueworker.go

```diff
@@ -0,0 +1,278 @@
+package actpool
+
+import (
+	"context"
+	"encoding/hex"
+	"math/big"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/iotexproject/go-pkgs/cache/ttl"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-core/action"
+	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/tracer"
+)
+
+type (
+	queueWorker struct {
+		queue chan workerJob
+		ap    *actPool
+
+		mu            sync.RWMutex
+		accountActs   map[string]ActQueue
+		emptyAccounts *ttl.Cache
+	}
+
+	workerJob struct {
+		ctx context.Context
+		act action.SealedEnvelope
+		err chan error
+	}
+
+	pendingActions struct {
+		sender string
+		acts   []action.SealedEnvelope
+	}
+)
+
+func newQueueWorker(ap *actPool, jobQueue chan workerJob) *queueWorker {
+	acc, _ := ttl.NewCache()
+	return &queueWorker{
+		queue:         jobQueue,
+		ap:            ap,
+		accountActs:   make(map[string]ActQueue),
+		emptyAccounts: acc,
+	}
+}
+
+func (worker *queueWorker) Start() {
+	if worker.queue == nil || worker.ap == nil {
+		panic("worker is invalid")
+	}
+	for {
+		job, more := <-worker.queue
```

Comment:
in addition to above comment, we also need a `Stop()` to close the `worker.queue`? otherwise the for loop will block here, waiting to receive more from the channel

Reply:
add a TODO for this enhancement